src/api/services/iuf/iuf.go

```diff
@@ -73,22 +74,29 @@ type IufService interface {
 
 // IufService service layer
 type iufService struct {
-	logger           utils.Logger
-	workflowCient    workflow.WorkflowServiceClient
-	k8sRestClientSet kubernetes.Interface
-	keycloakService  services_shared.KeycloakService
-	env              utils.Env
+	logger                 utils.Logger
+	workflowClient         workflow.WorkflowServiceClient
+	workflowTemplateClient workflowtemplate.WorkflowTemplateServiceClient
+	k8sRestClientSet       kubernetes.Interface
+	keycloakService        services_shared.KeycloakService
+	env                    utils.Env
 }
 
 // NewIufService creates a new Iufservice
 func NewIufService(logger utils.Logger, argoService services_shared.ArgoService, k8sSvc services_shared.K8sService, keycloakService services_shared.KeycloakService, env utils.Env) IufService {
 
+	workflowTemplateClient, err := argoService.Client.NewWorkflowTemplateServiceClient()
```

Comment:
Is there a reason this isn't passed in as a parameter to `NewIufService` and then handled with dependency injection withthe `fx` module?

Edit: oh, I see it uses the `argoService` that gets passed in, and that is also used to call `NewWorkflowServiceClient`, so this is following that pattern. I do still wonder if it should/could be handled with `fx`.

Reply:
It could be handled with fx but then I think it would be better to not pass argoService, and instead provide its constituents. Though I do think it is harder to figure out what pieces of an imported dependency are provided vs not.